dagger/stores/memorystore.nim

```diff
@@ -7,54 +7,72 @@
 ## This file may not be copied, modified, or distributed except according to
 ## those terms.
 
+{.push raises: [Defect].}
+
 import std/sequtils
+
 import pkg/chronos
 import pkg/libp2p
+import pkg/chronicles
+import pkg/questionable
 
-import ../stores/blockstore
+import ./blockstore
 import ../blocktype
 
 export blockstore
 
+logScope:
+  topics = "dagger memstore"
+
 type
   MemoryStore* = ref object of BlockStore
-    blocks: seq[Block] # TODO: Should be an LRU cache
+    blocks: seq[?Block] # TODO: Should be an LRU cache
```

Comment:
Why would you store a None block?

Reply:
This is a bit of a placeholder and should be replaced with a proper LRU cache, mostly for convenience so I don't have to wrap it in a `some/none` call everywhere.